java-util/src/main/java/io/druid/java/util/common/concurrent/Execs.java

```diff
@@ -49,6 +51,11 @@ public static ExecutorService dummy()
     return DummyExecutorService.INSTANCE;
   }
 
+  public static ListeningExecutorService sameThreadExecutor()
```

Comment:
It was a mistake in Guava's API to call this method this way, because it gives full impression that some static constant is returned from it (I thought that until now), while in reality this method (and the removed method in Guava) creates a new Object each time. (Guava developers fixed this mistake by calling the new method *new*DirectExecutorService().)

However, I think that Druid actually never tries to terminate this service, so it would be better to keep the method name as is, actually cache the executor service, and simplify it's implementation, by either throwing an `UnsupportedOperationException` in `shutdown()` and `shutdownNow()` or making them no-ops, and always returning `false` from `isTerminated()`, and removing all state and synchronization from the class.

Reply:
That's reasonable

Reply:
fixed